packages/elements-core/src/utils/ref-resolving/resolvedObject.test.ts

```diff
@@ -236,4 +236,176 @@ describe('createResolvedObject', () => {
+          'x-stoplight': { 'error-message': 'You do not have permission to view this reference', id: 'nezai0hyj4yak' },
+        },
+        { $ref: '#/__bundled__/iq2mwk8jvthd2', 'x-stoplight': { id: 'ovj32wmpxpg7p' } },
+      ],
+      'x-stoplight': { id: 'b73ff5df9864f' },
+    };
+
+    const resolvedObject = getOriginalObject(originalObject);
+    const filteredObject = {
+      anyOf: [
+        { $ref: '#/__bundled__/0mui9s02880hl', 'x-stoplight': { id: '19c178fc05d4a' } },
+        { $ref: '#/__bundled__/iq2mwk8jvthd2', 'x-stoplight': { id: 'ovj32wmpxpg7p' } },
+      ],
+      'x-stoplight': { id: 'b73ff5df9864f' },
+    };
+    expect(resolvedObject).toEqual(filteredObject);
+  });
+
+  it('show the error if all object contains an error for anyOf', () => {
+    const originalObject = {
+      anyOf: [
+        [
+          {
+            'x-sl-error-message': 'You do not have permission to view this reference',
+            'x-stoplight': {
+              'error-message': 'You do not have permission to view this reference',
+              id: 'nezai0hyj4yak',
+            },
+          },
+          {
+            'x-sl-error-message': 'You do not have permission to view this reference',
+            'x-stoplight': {
+              'error-message': 'You do not have permission to view this reference',
+              id: 'nezai0hyj4yak',
+            },
+          },
+          {
+            'x-sl-error-message': 'You do not have permission to view this reference',
+            'x-stoplight': {
+              'error-message': 'You do not have permission to view this reference',
+              id: 'nezai0hyj4yak',
+            },
+          },
+        ],
+      ],
+      'x-stoplight': { id: 'b73ff5df9864f' },
+    };
+
+    const resolvedObject = getOriginalObject(originalObject);
+    const filteredObject = {
+      anyOf: [
+        [
+          {
+            'x-sl-error-message': 'You do not have permission to view this reference',
+            'x-stoplight': {
+              'error-message': 'You do not have permission to view this reference',
+              id: 'nezai0hyj4yak',
+            },
+          },
+          {
+            'x-sl-error-message': 'You do not have permission to view this reference',
+            'x-stoplight': {
+              'error-message': 'You do not have permission to view this reference',
+              id: 'nezai0hyj4yak',
+            },
+          },
+          {
+            'x-sl-error-message': 'You do not have permission to view this reference',
+            'x-stoplight': {
+              'error-message': 'You do not have permission to view this reference',
+              id: 'nezai0hyj4yak',
+            },
+          },
+        ],
+      ],
+      'x-stoplight': { id: 'b73ff5df9864f' },
+    };
+    expect(resolvedObject).toEqual(filteredObject);
+  });
+  it('show the error if all object contains an error for oneOf', () => {
```

Comment:
may be you can put a comment saying "The schema marked as internal so it's populate the error object.